fix(cta-block): fix medium breakpoint layout and spacing

mwanberg opened the pull request:
### Related Ticket(s)

DDS issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7176
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2234

### Description

This PR fixes the medium breakpoint grid columns count and related spacing. It also removes a now-unneeded SCSS file.

### Testing Instructions

- Visit the `CTA Block` component 
  - Default: `?path=/story/components-cta-block--default`
  - With content items: `?path=/story/components-cta-block--with-content-items`
  - With link list: `?path=/story/components-cta-block--with-link-list`
- Check the layout and spacing at all breakpoints to make sure the component has the correct column count and spacing.

oliviaflory commented:
Hey @mwanberg this looks really close! I'm seeing a few shifts in the Percy build:

### 1. CTA block with content items
The 2nd and 3rd content items are not aligned to the grid, see expected alignment below:

**PR**
<img width="1312" alt="Screen Shot 2021-10-15 at 12 19 09 PM" src="https://user-images.githubusercontent.com/45692622/137523705-8fc210e7-d457-4ba3-9877-841eca24b38f.png">

**Expected**
<img width="1312" alt="Screen Shot 2021-10-15 at 12 18 56 PM" src="https://user-images.githubusercontent.com/45692622/137523689-b08b580a-f024-4c73-b2ff-9b84c62f6779.png">




### 2. CTA block with link list 
The link list has shifted to the left a bit, see the expected grid alignment below

**PR**
<img width="1311" alt="Screen Shot 2021-10-15 at 12 43 43 PM" src="https://user-images.githubusercontent.com/45692622/137523784-b70a5dd7-a879-4ca1-bf88-9a70a754f8c1.png">

**Expected**
<img width="1312" alt="Screen Shot 2021-10-15 at 12 42 26 PM" src="https://user-images.githubusercontent.com/45692622/137523777-bda12566-27b5-4a83-a889-7da92b098da1.png">

### 3. CTA section with link list
The link list is not extending the full width on the small breakpoint, looks like it got inset a bit
<img width="1178" alt="Screen Shot 2021-10-15 at 12 55 49 PM" src="https://user-images.githubusercontent.com/45692622/137524961-1e429104-a2c7-40ba-851f-911b92df0cfa.png">



### 4. CTA section
I'm seeing some line breaks in the CTA section on the small breakpoints, I'm not sure if anything was changed in this PR on the width, but seems similar to what is happening with the link list (above).
<img width="1049" alt="Screen Shot 2021-10-15 at 1 13 25 PM" src="https://user-images.githubusercontent.com/45692622/137527174-62ebf3ec-2620-41ee-ac1b-b9ad7a23e4a4.png">

![CTA-section-mobile--Oct-15-2021 13-17-23](https://user-images.githubusercontent.com/45692622/137527475-970daab7-ab4f-4812-a3fe-29e8474b0ea0.gif)

mwanberg commented:
@oliviaflory Thank you for reviewing! I removed the right padding adjustment that was causing the grid alignment issues for CTA Block. The changes we saw in the CTA section were because there was extra padding getting applied for the embedded CTA Block in the CTA Section styles to compensate for some padding that was originally missing (which the PR fixed). I removed the extra padding in there, so now I think everything should line up as it was before.
